rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,417 @@
+            fee_asset (Asset): fee asset
+            sender (ChecksumEvmAddress): sender address
+        """
+        return self.base.make_event(
+            tx_hash=tx_hash,
+            sequence_index=self.base.get_next_sequence_index(),
+            timestamp=timestamp,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=fee_asset,
+            amount=(fee_amount := asset_normalized_value(amount=fee_raw, asset=fee_asset)),
+            location_label=sender,
+            notes=f'Spend {fee_amount} {fee_asset.resolve_to_asset_with_symbol().symbol} as Rainbow Swap fees',  # noqa: E501
+            counterparty=CPT_RAINBOW_SWAPS,
+            address=self.router_address,
+        )
+
+    def _maybe_decode_swap_native_to_token(
+            self,
+            transaction: EvmTransaction,
+            event: 'EvmEvent',
+            decoded_events: list['EvmEvent'],
+            sender: ChecksumEvmAddress,
+            wrap_event: EvmTxReceiptLog,
+    ) -> None:
+        """
+        Decodes a swap transaction from native token to another token.
+        Check if the first log is deposit and the last is transfer log
+        """
+        # The fee amount is the difference of msg.value and the wrapped token amount
+        try:
+            swap_amount = asset_raw_value(event.amount, self.evm_inquirer.native_token)
+        except UnsupportedAsset:
+            log.error(
+                f'Failed to decode swap from native token to token'
+                f' for transaction {transaction.tx_hash.hex()} '
+                f'due to unsupported asset {event.asset.identifier}.',
+            )
+            return
+
+        fee_raw = swap_amount - int.from_bytes(wrap_event.data)
+
+        # Create fee event
+        fee_event = self._create_fee_event(
+            tx_hash=transaction.tx_hash,
+            timestamp=transaction.timestamp,
+            fee_raw=fee_raw,
+            fee_asset=self.evm_inquirer.native_token,
+            sender=sender,
+        )
+        decoded_events.append(fee_event)
+
+        return
+
+    def _maybe_decode_fee_from_swap_token_to_native(
+            self,
+            transaction: EvmTransaction,
+            decoded_events: list['EvmEvent'],
+            in_event: Optional['EvmEvent'],
+            out_event: Optional['EvmEvent'],
+            sender: ChecksumEvmAddress,
+            unwrap_event: EvmTxReceiptLog,
+    ) -> None:
+        """
+        Decodes the fee from a swap transaction involving token to native token.
+        Via the first transfer log transfer and the last withdrawn log
+        """
+        native_token = self.evm_inquirer.native_token
+
+        if in_event is None or out_event is None:
+            log.error(
+                f'Failed to decode fee from swap token to native token '
+                f'for transaction {transaction.tx_hash.hex()}.',
+            )
```

Comment:
use the str version of transaction, not just the hash. The reason as I said is that it displays the chain that is important to find the transaction